pkg/credentialprovider/azure_credentials.go

```diff
@@ -232,24 +240,55 @@ func (a *acrProvider) getFromACR(ctx context.Context, loginServer string) (strin
 // parseACRLoginServerFromImage takes image as parameter and returns login server of it.
 // Parameter `image` is expected in following format: foo.azurecr.io/bar/imageName:version
 // If the provided image is not an acr image, this function will return an empty string.
-func (a *acrProvider) parseACRLoginServerFromImage(image string) string {
-	match := acrRE.FindAllString(image, -1)
+func (a *acrProvider) parseACRLoginServerFromImage(image string) (string, string) {
+	processedImage, mirrorRegistry := a.processImageWithMirrorMapping(image)
+
+	match := acrRE.FindAllString(processedImage, -1)
 	if len(match) == 1 {
-		return match[0]
+		return match[0], mirrorRegistry
 	}
 
 	// handle the custom cloud case
 	if a != nil && a.environment != nil {
 		cloudAcrSuffix := a.environment.ContainerRegistryDNSSuffix
 		cloudAcrSuffixLength := len(cloudAcrSuffix)
 		if cloudAcrSuffixLength > 0 {
-			customAcrSuffixIndex := strings.Index(image, cloudAcrSuffix)
+			customAcrSuffixIndex := strings.Index(processedImage, cloudAcrSuffix)
 			if customAcrSuffixIndex != -1 {
 				endIndex := customAcrSuffixIndex + cloudAcrSuffixLength
-				return image[0:endIndex]
+				return processedImage[0:endIndex], mirrorRegistry
 			}
 		}
 	}
 
-	return ""
+	return "", ""
+}
+
+// acrProvider mirror mapping example: {"mcr.microsoft.com": "abc.azurecr.io"}
+// processImageWithMirrorMapping input format: "mcr.microsoft.com/bar/image:version"
+// output format: "abc.azurecr.io/bar/image:version", "mcr.microsoft.com"
+func (a *acrProvider) processImageWithMirrorMapping(image string) (string, string) {
+	for sourceRegistry, targetRegistry := range a.mirrorMapping {
+		if strings.HasPrefix(image, sourceRegistry) {
+			return strings.Replace(image, sourceRegistry, targetRegistry, 1), sourceRegistry
+		}
+	}
+	return image, ""
+}
+
+// processMirrorMapping input format: "--mirror-mapping=aaa:bbb,ccc:ddd"
```

Comment:
how to handle "aaa:bbb,aaa:ccc"?

Reply:
Current logic takes `aaa:ccc`

Reply:
so this PR only supports one mirror mapping?

Reply:
No, it supports multiple.
Bin's question is how to handle two mappings with the same source registry. The logic is that the latter one takes effect.